File: pyg/pyg-shop-web/src/main/webapp/js/service/specificationService.js
//定义服务
app.service("specificationService",function ($http) {
    //查询所有规格
    this.findAll=function () {
        return $http.get("../specification/findAll");
    };
    //分页查询的方法
    this.findPage=function (page,rows) {
        return $http.get("../specification/findPage/"+page+"/"+rows);
    };
    //添加规格方法
    this.add=function (entity) {
        return $http.post("../specification/save",entity);
    };
    //更新规格的方法
    this.update=function (entity) {
        return $http.post("../specification/update",entity);
    };
    //删除规格方法
    this.dele=function (ids) {
        return $http.get("../specification/delete/"+ids);
    };
    //根据id查询规格数据
    this.findOne=function (id) {
        return $http.get("../specification/findOne/"+id);
    };
    //定义搜索方法
    this.searchResult=function (page, rows,query) {
        return $http.post("../specification/findQv?page="+page+"&rows="+rows,query);
    };
    //定义查询规格显示下拉款方法
    this.findSpecList=function () {
        return $http.get("../specification/findSpecList");
    }
});

File: pyg/pyg-manager/pyg-manager-interface/src/main/java/com/pyg/manager/service/SpecificationService.java
package com.pyg.manager.service;

import com.pyg.pojo.TbSpecification;
import com.pyg.utils.PageResult;
import com.pyg.vo.Specifiaction;

import java.util.List;
import java.util.Map;

/**
 * @Author: Mr.jia
 * @Date: 2018/9/22 21:22
 */
public interface SpecificationService {
	/**
	 * 查询所有规格
	 */
	public List<TbSpecification> findAll();

	/**
	 * 需求:分页查询所有规格
	 */
	public PageResult findPage(Integer page, Integer rows);

	/**
	 * 需求:添加规格
	 */
	public void insert(Specifiaction specifiaction);

	/**
	 * 需求:根据id修改
	 */
	public Specifiaction findOne(long id);

	/**
	 * 需求:根据id更新规格数据
	 */
	public void update(Specifiaction specifiaction);

	/**
	 * 需求:更具id删除规格数据
	 */
	public void delete(Long[] ids);

	/**
	 * 需求:查询规格数据进行多项选择
	 */
	public List<Map> findSpecList();
}
